superset/common/query_context_processor.py

```diff
@@ -461,7 +464,17 @@ def get_data(self, df: pd.DataFrame) -> Union[str, List[Dict[str, Any]]]:
                     df, index=include_index, **config["CSV_EXPORT"]
                 )
             elif self._query_context.result_format == ChartDataResultFormat.XLSX:
-                result = excel.df_to_excel(df, **config["EXCEL_EXPORT"])
+                ndf = df.copy()
+                columns = list(df)
+                i = 0
+                for col in columns:
+                    coltype = None
+                    if i < len(coltypes):
+                        coltype = coltypes[i]
+                        if coltype == GenericDataType.NUMERIC:
+                            ndf[col] = pd.to_numeric(df[col], errors="ignore")
```

Comment:
Maybe we can leave this for a follow-up if it's needed, but do you think it would make sense to apply temporal formatting for `GenericDataType.TEMPORAL`?
```suggestion
                        if coltype == GenericDataType.NUMERIC:
                            ndf[col] = pd.to_numeric(df[col], errors="ignore")
```

Reply:
I wondered, and from some manual tests with libreoffice and excel, I got better result leaving things as is. But of course it would require much more testing (with timezones and locales and what not), things I cannot do. 

Leaving this for other volunteers would suit me to be honest ;)